### PR TITLE
Added rounding method for conversion_ratio

### PR DESCRIPTION
**Pull request by daviidy**
# ASK
For issue #21 

I'm not sure of the `"NORMAL"` method. I left this question in Slack
Screenshot 2023-05-10 at 11 19 58 AM

### CODE REVIEW

File: test/model.test.ts [hunk 1]
```diff
@@ -203,6 +204,7 @@ describe(OCX.Model, () => {
                   numerator: opts.to,
                   denominator: opts.convertsFrom,
                 },
+                rounding_type: rounding_type,
```
Comment (pjohnmeyer) [inline test/model.test.ts:207]: You changed a test helper function but you never call it with this parameter so, why?

**Please fix:** Please write tests for the model.

File: test/workbook/stakeholder-sheet.test.ts [hunk 2]
```diff
@@ -314,4 +314,76 @@ describe(StakeholderSheet, () => {
...
+        conversion_ratio: 1.3,
+        rounding_type: "FLOOR",
+      },
+      {
+        display_name: "Class A Preferred Stock Ceiling",
+        is_preferred: true,
+        conversion_ratio: 0.7,
+        rounding_type: "CEILING",
+      }
+    );
```
Comment (pjohnmeyer) [inline test/workbook/stakeholder-sheet.test.ts:352]: **Non-blocking:** There's a lot more setup here than is necessary to test this; the purpose of the test is obfuscated by all of the unnecessary classes.

I think this test could have safely lived in a much smaller more focused test in `holdings-column.test.ts`. It isn't particularly relevant to the larger "sheet".

File: src/workbook/stakeholder-sheet/holdings-columns.ts [hunk 3]
```diff
@@ -149,10 +149,32 @@ export class StockClassAsConvertedColumn {
...
 
+    const conversion_value = `${
+      outstandingRange.getExtents().topLeftAddress
+    } * ${stockClass.conversion_ratio}`;
+
+    enum RoundingType {
+      CEILING = "CEILING",
+      FLOOR = "FLOOR",
+      NORMAL = "NORMAL",
+    }
```
Comment (pjohnmeyer) [inline src/workbook/stakeholder-sheet/holdings-columns.ts:160]: **Question:** What is the purpose of defining this type if it is used only in this method?

File: src/workbook/stakeholder-sheet/holdings-columns.ts [hunk 4]
```diff
@@ -149,10 +149,32 @@ export class StockClassAsConvertedColumn {
...
+        formula = `CEILING(${conversion_value}, 1)`;
+        break;
+      case RoundingType.FLOOR:
+        formula = `FLOOR(${conversion_value}, 1)`;
+        break;
+      case RoundingType.NORMAL:
+      default:
+        formula = `ROUND(${conversion_value}, 0)`;
+        break;
+    }
```
Comment (pjohnmeyer) [inline src/workbook/stakeholder-sheet/holdings-columns.ts:174]: **Non-blocking:** This looks like a good candidate for some functional decomposition.

File: src/model/index.ts [hunk 5]
```diff
@@ -318,7 +323,14 @@ class Model implements WorkbookModel {
...
     // look at putting the `Big` types directly on the interface to avoid
     // precision loss.
-    return Calculations.convertRatioToDecimalNumber(mechanism.ratio).toNumber();
+    return Calculations.convertRatioToDecimalNumber(mechanism.ratio);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private getStockClassRoundingType(value: any): string {
+    const mechanism = Array.of(value).flat()[0]?.conversion_mechanism;
+    const roundingType = mechanism?.rounding_type || "NORMAL";
```
Comment (pjohnmeyer) [inline src/model/index.ts:332]: 👍 Good job handling the default / missing data case.

**Review by pjohnmeyer — CHANGES_REQUESTED** [review]

File: src/workbook/stakeholder-sheet/holdings-columns.ts (hunk 3 above)
Comment (daviidy) [reply src/workbook/stakeholder-sheet/holdings-columns.ts:160]: It's not necessary actually. Just a bad habit of abstracting everything lol

**Review by pjohnmeyer — APPROVED** [review]

**Pull request merged**